DataFormats/EgammaCandidates/interface/GsfElectron.h

```diff
@@ -35,443 +33,418 @@ namespace reco
+    float deltaPhiSuperClusterTrackAtVtx() const { return trackClusterMatching_.deltaPhiSuperClusterAtVtx; }
+    float deltaPhiSeedClusterTrackAtCalo() const { return trackClusterMatching_.deltaPhiSeedClusterAtCalo; }
+    float deltaPhiEleClusterTrackAtCalo() const { return trackClusterMatching_.deltaPhiEleClusterAtCalo; }
+    float deltaEtaSeedClusterTrackAtVtx() const {
+      return superCluster().isNonnull() && superCluster()->seed().isNonnull()
+                 ? trackClusterMatching_.deltaEtaSuperClusterAtVtx - superCluster()->eta() +
+                       superCluster()->seed()->eta()
+                 : std::numeric_limits<float>::max();
     }
-    const TrackClusterMatching & trackClusterMatching() const { return trackClusterMatching_ ; }
+    const TrackClusterMatching &trackClusterMatching() const { return trackClusterMatching_; }
 
     // for backward compatibility, usefull ?
-    void setDeltaEtaSuperClusterAtVtx( float de ) { trackClusterMatching_.deltaEtaSuperClusterAtVtx = de ; }
-    void setDeltaPhiSuperClusterAtVtx( float dphi ) { trackClusterMatching_.deltaPhiSuperClusterAtVtx = dphi ; }
-
+    void setDeltaEtaSuperClusterAtVtx(float de) { trackClusterMatching_.deltaEtaSuperClusterAtVtx = de; }
+    void setDeltaPhiSuperClusterAtVtx(float dphi) { trackClusterMatching_.deltaPhiSuperClusterAtVtx = dphi; }
 
   private:
-
     // attributes
-    TrackClusterMatching trackClusterMatching_ ;
-
-
-  //=======================================================
-  // Track extrapolations
-  //=======================================================
+    TrackClusterMatching trackClusterMatching_;
 
-  public :
+    //=======================================================
+    // Track extrapolations
+    //=======================================================
 
-    struct TrackExtrapolations
-     {
-      math::XYZPointF  positionAtVtx ;     // the track PCA to the beam spot
-      math::XYZPointF  positionAtCalo ;    // the track PCA to the supercluster position
-      math::XYZVectorF momentumAtVtx ;     // the track momentum at the PCA to the beam spot
-      math::XYZVectorF momentumAtCalo ;    // the track momentum extrapolated at the supercluster position from the innermost track state
-      math::XYZVectorF momentumOut ;       // the track momentum extrapolated at the seed cluster position from the outermost track state
-      math::XYZVectorF momentumAtEleClus ; // the track momentum extrapolated at the ele cluster position from the outermost track state
-      math::XYZVectorF momentumAtVtxWithConstraint ;     // the track momentum at the PCA to the beam spot using bs constraint
-     } ;
+  public:
+    struct TrackExtrapolations {
+      math::XYZPointF positionAtVtx;   // the track PCA to the beam spot
+      math::XYZPointF positionAtCalo;  // the track PCA to the supercluster position
+      math::XYZVectorF momentumAtVtx;  // the track momentum at the PCA to the beam spot
+      math::XYZVectorF
```

Comment:
Just to notice a few line splits because of a comment at the end of the line.
I am not particularly against them here, though